src/utils/keybindings.ts

```diff
@@ -0,0 +1,172 @@
+// kilocode_change - new file
+import packageJson from "../package.json"
+import { readUserConfigFile } from "./vscode-config"
+
+export type KeybindingEntry = {
+	key?: string
+	command?: string
+	when?: string
+	args?: unknown
+	mac?: string
+}
+
+// Module-level constants for performance (created once, not per call)
+const MAC_MODIFIERS: Record<string, string> = {
+	cmd: "Cmd",
+	meta: "Cmd",
+	ctrl: "Ctrl",
+	alt: "Option",
+	option: "Option",
+	shift: "Shift",
+}
+
+const WIN_LINUX_MODIFIERS: Record<string, string> = {
+	cmd: "Win",
+	meta: "Win",
+	ctrl: "Ctrl",
+	alt: "Alt",
+	shift: "Shift",
+}
+
+const SPECIAL_KEYS: Record<string, string> = {
+	left: "Left",
+	right: "Right",
+	up: "Up",
+	down: "Down",
+	home: "Home",
+	end: "End",
+	pageup: "PageUp",
+	pagedown: "PageDown",
+	insert: "Insert",
+	delete: "Delete",
+	backspace: "Backspace",
+	tab: "Tab",
+	enter: "Enter",
+	escape: "Escape",
+	space: "Space",
+}
+
+/**
+ * Gets the current keybinding for a list of commands, reading from user's keybindings.json
+ * and falling back to the extension's default keybinding from package.json
+ */
+export async function getKeybindingsForCommands(commandIds: string[]): Promise<Record<string, string>> {
+	const keybindings: Record<string, string> = {}
+
+	for (const commandId of commandIds) {
+		const keybinding = await getKeybindingForCommand(commandId)
+		if (keybinding) {
+			keybindings[commandId] = keybinding
+		}
+	}
+
+	return keybindings
+}
+
+export async function getKeybindingForCommand(commandId: string): Promise<string | undefined> {
+	try {
+		const userKeybindings = await readUserConfigFile("keybindings.json")
+		const userBinding = findFirstBindingForCommand(userKeybindings as KeybindingEntry[], commandId)
+
+		if (userBinding?.key) {
+			const rawKey = userBinding.key.trim()
+			if (rawKey) return prettyPrintKey(rawKey, process.platform)
+		}
+
+		return getDefaultKeybindingForCommand(commandId)
+	} catch (error) {
+		return undefined
+	}
+}
+
+/**
+ * Gets the default keybinding for a command from package.json
+ * @param commandId The command ID to look up
+ * @returns The platform-specific keybinding string (guaranteed to exist for valid commands)
+ * @throws Error if the command is not found in package.json keybindings
+ */
+export function getDefaultKeybindingForCommand(commandId: string): string {
+	const keybindings = packageJson.contributes?.keybindings as KeybindingEntry[] | undefined
+	const binding = keybindings?.find((kb) => kb.command === commandId)
+	if (!binding) {
+		throw new Error(`Command '${commandId}' not found in package.json keybindings`)
```

Comment:
given the purpose of this functionality (showing the user a key), I would say return null or 'not found' would be better. If we want to detect it we can still send an event to posthog, but it doesnt seem important enough to throw na exception right?